SICP_book/chapter_2_4.metta

```diff
@@ -0,0 +1,721 @@
+; This chapter is dedicated to generic procedures which works with different object representation, type tags and data-
+; directed programming.
+
+; Complex numbers.
+
+; Two different representations for complex numbers - (real, imag) parts and (magnitude, angle).
+
+; One way to implement complex numers' selectors:
+; (all sqrt/square/cos/atan etc functions are builtin in Scheme. I'll use imports from python)
+
+(= (square $x)
+    (* $x $x))
+
+!(import! &self additional_funcs)
+
+(= (sqrt $x)
+    (sqrt! $x))
+
+(= (atan $x $y)
+    (atan! $x $y))
+
+(= (cos $x)
+    (cos! $x))
+
+(= (sin $x)
+    (sin! $x))
+
```

Comment:
Not sure why do you need to duplicate these functions.
```suggestion

(= (sqrt $x)
    (sqrt! $x))

(= (atan $x $y)
    (atan! $x $y))

(= (cos $x)
    (cos! $x))

(= (sin $x)
    (sin! $x))

```

Reply:
Done.